samples/csharp/MicrosoftEduGraphSamples/Workflows/SubmissionWorkflow.cs

```diff
@@ -168,5 +202,109 @@ public async Task BatchRequestWorkflow()
                 Console.WriteLine($"BatchRequestWorkflow: {ex.ToString()}");
             }
         }
+
+        /// <summary>
+        /// A code sample to create a submission feedback resource
+        /// </summary>
+        public async Task SubmissionFeedbackResource(bool appOnly = false)
+        {
+            const int MAX_RETRIES = 10;
+
+            Console.WriteLine($"Create submission feedback resource");
+            // Get a Graph client based on the appOnly parameter
+            var graphClient = appOnly ? GraphClient.GetApplicationClient(_config["tenantId"], _config["appId"], _config["secret"]) : GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+            //Create new assigment
+            var assignment = await Assignment.CreateSampleAsync(graphClient, _config["classId"]);
+            var assignmentId = assignment.Id;
+            Console.WriteLine($"Assignment created {assignmentId}");
+
+            await Assignment.SetUpAssignmentFeedbackResourcesFolderAsync(graphClient, _config["classId"], assignmentId);
+            Console.WriteLine("SetupResourceFolder creation successful");
+
+            // Check feedback resource folder
+            int retryNum = 0;
+            while (assignment.FeedbackResourcesFolderUrl == null && retryNum <= MAX_RETRIES)
+            {
+                assignment = await Assignment.GetAssignmentAsync(graphClient, _config["classId"], assignmentId);
+                retryNum++;
+            }
+            Console.WriteLine($"Feedback resources folder: {assignment.FeedbackResourcesFolderUrl}");
+
+            //Publish assignment
+            assignment = await GlobalMethods.PublishAssignmentsAsync(graphClient, assignment.Id);
+            Thread.Sleep(7000);
```

Comment:
why do we have samples with flat Thread.Sleep patterns?
We should do a publish status check for a minute or so and wrap this whole waiting inside a function so here it will read something like AssignmentAssignedCheckAsync(, timeout = 60sec)

Reply:
Removing thread.sleep as the logic is implemented in "PublishAssignmentsAsync" from globalmethods.cs file.